src/hist_source.cc

```diff
@@ -73,7 +73,9 @@ void hist_source::text_value_for_line(textview_curses &tc,
 
         bucket_index          = brow - 1;
         group_iter = this->hs_groups.begin();
-        advance(group_iter, grow);
+        size_t groups_size = (size_t)this->hs_groups.size();
```

Comment:
The 'grow' value (group-row) should not be outside the range of the hs_groups map.  So, it seems more like the 'row' value coming into the function is out-of-range, which is probably the actual bug.